PSICompile/ILCodeGen.cs

```diff
@@ -66,9 +85,34 @@ public class ILCodeGen : Visitor {
       }
       if (w.NewLine) Out ("    call void [System.Console]System.Console::WriteLine ()");
    }
-   
-   public override void Visit (NIfStmt f) => throw new NotImplementedException ();
-   public override void Visit (NForStmt f) => throw new NotImplementedException ();
+
+   public override void Visit (NIfStmt f) {
+      string lab1 = NextLabel (), lab2 = NextLabel (), lab3 = NextLabel ();
+      Out ($"    br {lab2}");
+      Out ($"  {lab1}:"); f.IfPart.Accept (this); Out ($"    br {lab3}");
+      Out ($"  {lab2}:"); f.Condition.Accept (this); 
+      Out ("    ldc.i4.1\n    ceq");
+      Out ($"    brtrue {lab1}");
+      if (f.ElsePart != null) f.ElsePart.Accept (this);
+      Out ($"  {lab3}:"); Out ("  nop");

```

Comment:
For an `if` statement, you don't need 3 labels, and you don't need to jump to the end to evaluate the condition. That is useful for while and repeat loops, because if the condition fails, you just want to continue to the next line. In an if statement, you can reduce this to 2 labels, and start with this first line:
`f.Condition.Accept (this);`

Reply:
Simplified as suggested